### PR TITLE
Add Stardoc

### PR DESCRIPTION
**Pull request by pauldraper**
Stardoc is the undocumented, future, non-fragile form of Skydoc.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -32,16 +32,16 @@ jobs:
     - stage: build
       <<: *osx
       env: V=0.16.1
-      script: ./travis.sh build
+      script: ./travis.sh build-sans-doc
```
Comment (andyscott) [inline .travis.yml]: Is it possible to build the docs automatically? Then we can have CI do a `git diff` check?

Comment (pauldraper) [reply .travis.yml]: We do a `git diff` check.

https://github.com/andyscott/rules_scala_annex/blob/282786d32c42c473d712c83df254dc482eea6011/travis.sh#L20

However, the stardoc binary is in \@io_bazel and the version used is only compatible with Bazel 0.18. So building the doc fails for any version except 0.18.

Comment (jjudd) [reply .travis.yml]: When did they move stardoc into io_bazel? I assume then, that the version in the skydoc repo is very out of date. TIL.

Comment (pauldraper) [reply .travis.yml]: The Stardoc binary is and always has been in the bazel repo. It is the legitimate Skylark interpreter.

The skydoc repo has the rules for using Skydoc (the janky Python-based thing) and Stardoc.

https://github.com/bazelbuild/skydoc/issues/102

**Pull request merged**